team-telnyx/telnyx-webrtc-ios
# WEBRTC-413: Implement Unit Tests

[WebRTC-413 - [iOS] Implement Unit Tests](https://telnyx.atlassian.net/browse/WEBRTC-413)

---


Implemented general unit tests. Flows covered:
- Socket connection and disconnection.
- Login with wrong credentials.
- Login with valid credentials (Require to replace constants in TestConstants.swift file)
- Peer related tests
- Call related tests.

## :older_man: :baby: Behaviors

### Before changes
The SDK had no Unit Testing

### After changes
SDK has Unit Testing


## ✋ Manual testing

1. Build the SDK
<img width="383" alt="Screen Shot 2021-03-19 at 15 00 28" src="https://user-images.githubusercontent.com/75636882/111823649-f5df8980-88c3-11eb-9e69-470b4fd8da0c.png">

2. Setup valid credentials into **TestConfigurations** file 
<img width="502" alt="Screen Shot 2021-03-19 at 15 05 42" src="https://user-images.githubusercontent.com/75636882/111824167-9afa6200-88c4-11eb-9fb2-87c2f6da6481.png">

3. Run the Tests
<img width="405" alt="Screen Shot 2021-03-19 at 15 00 48" src="https://user-images.githubusercontent.com/75636882/111823671-fb3cd400-88c3-11eb-9fa7-9f82f606de41.png">

4. All tests should run. There's 1 unit test that is going to fail **testSocketBasics()**. This is because there's an error on the code that has to be fixed (Check [WEBRTC-435](https://telnyx.atlassian.net/browse/WEBRTC-435))
